output/socket/outputsocket.go

```diff
@@ -0,0 +1,70 @@
+package socket
+
+import (
+	"context"
+
+	"github.com/tsaikd/gogstash/config"
+	"github.com/tsaikd/gogstash/config/logevent"
+	"golang.org/x/sync/errgroup"
+	"net"
+)
+
+// ModuleName is the name used in config file
+const ModuleName = "socket"
+
+// OutputConfig holds the configuration json fields and internal objects
+type OutputConfig struct {
+	config.OutputConfig
+	Socket       string `json:"socket"`  // Type of socket, must be one of ["tcp", "unix", "unixpacket"].
+	Address      string `json:"address"` // For TCP, address must have the form `host:port`. For Unix networks, the address must be a file system path.
+	outputSocket *net.Conn
+}
+
+// DefaultOutputConfig returns an OutputConfig struct with default values
+func DefaultOutputConfig() OutputConfig {
+	return OutputConfig{
+		OutputConfig: config.OutputConfig{
+			CommonConfig: config.CommonConfig{
+				Type: ModuleName,
+			},
+		},
+	}
+}
+
+// InitHandler initialize the output plugin
+func InitHandler(ctx context.Context, raw *config.ConfigRaw) (config.TypeOutputConfig, error) {
+	conf := DefaultOutputConfig()
+	if err := config.ReflectConfig(raw, &conf); err != nil {
+		return nil, err
+	}
+
+	// init Socket
+	eg, ctx := errgroup.WithContext(ctx)
+	conn, err := net.Dial(conf.Socket, conf.Address)
+	if err != nil {
+		return nil, err
+	}
+	eg.Go(func() error {
+		select {
+		case <-ctx.Done():
+			return conn.Close()
+		}
+	})
```

Comment:
errgroup without Wait() ?
How about use native goroutine?

Reply:
Will change to native `go`